### PR TITLE
Filter attributes for listed users

### LINKED ISSUE #37: As a insurance agent with many clients, I want to filter and see particular details about my clients
so that I can easily check certain details such as policy ids or phone numbers of my clients without having to sieve through other details.

### PR DESCRIPTION
**Pull request by jay9645**
Not sure if the implementation is a little hacky, as I couldn't figure out a way to manipulate the UI without manipulating the list. Current implementation is to modify the list to show only the specified fields, then undo the modification before the next command. Can try to clone and test.

### CODE REVIEW

File: docs/UserGuide.md [hunk 1]
```diff
@@ -95,11 +95,16 @@ Examples:
...
 * (example with no policy)`add n/Betsy Crowe e/betsycrowe@example.com a/Kent Ridge t/nopolicy`
 
-### Listing all persons : `list`
+### Listing all persons : `list [-ATTRIBUTE]`
 
-Shows a list of all clients in ClientBook.
+Shows a list of all clients in ClientBook. An optional attribute option can be added to show a list with only the specified attribute.
 
-Format: `list`
+Format: `list [-ATTRIBUTE]`
```
Comment (wongkokian) [inline docs/UserGuide.md]: Would be nice if can add a screenshot of how the feature works.

File: src/main/java/seedu/address/logic/commands/ListCommand.java [hunk 2]
```diff
@@ -3,22 +3,62 @@
...
+ * Lists all persons in the address book to the user. Attribute can be specified to show only specific attributes.
  */
 public class ListCommand extends Command {
 
     public static final String COMMAND_WORD = "list";
 
-    public static final String MESSAGE_SUCCESS = "Listed all persons";
+    public static final String MESSAGE_SUCCESS = "Listed all clients";
+
+    public static final String MESSAGE_SUCCESS_ATTRIBUTE = "Listed all clients and filtered specified attribute";
```
Comment (wongkokian) [inline src/main/java/seedu/address/logic/commands/ListCommand.java]: Would it be possible for the success message to output the attribute that was specified as well? Like "Listed all clients with address attribute as filter."

File: src/main/java/seedu/address/logic/commands/ListCommand.java [hunk 3]
```diff
@@ -3,22 +3,62 @@
...
+    public boolean isAttributeSpecified() {
+        return this.attribute.isPresent();
+    }
 
     @Override
     public CommandResult execute(Model model) {
         requireNonNull(model);
-        model.updateFilteredPersonList(PREDICATE_SHOW_ALL_PERSONS);
-        return new CommandResult(MESSAGE_SUCCESS);
+        if (this.attribute.isEmpty()) {
```
Comment (wongkokian) [inline src/main/java/seedu/address/logic/commands/ListCommand.java]: A suggestion will be to do "if (!isAttributeSpecified()) {".

File: src/main/java/seedu/address/logic/parser/ParserUtil.java [hunk 4]
```diff
@@ -38,6 +39,26 @@ public static Index parseIndex(String oneBasedIndex) throws ParseException {
...
+            return Attribute.POLICY_ID;
+        } else if (trimmedAttribute.equals("-phone")) {
+            return Attribute.PHONE;
+        } else if (trimmedAttribute.equals("-email")) {
+            return Attribute.EMAIL;
+        } else if (trimmedAttribute.equals("-address")) {
+            return Attribute.ADDRESS;
+        } else {
+            throw new ParseException(Attribute.MESSAGE_CONSTRAINTS);
+        }
```
Comment (wongkokian) [inline src/main/java/seedu/address/logic/parser/ParserUtil.java]: One suggestion would be to use switch statements.

File: src/main/java/seedu/address/model/person/Person.java [hunk 5]
```diff
@@ -47,26 +49,53 @@ public Person(Name name, Phone phone, Email email, Address address, Set<Tag> tag
...
+            this.address = Optional.empty();
+        } else if (attribute.equals(Attribute.ADDRESS)) {
+            this.phone = Optional.empty();
+            this.email = Optional.empty();
+            this.address = Optional.of(person.getAddress().get());
+        } else {
+            this.phone = Optional.empty();
+            this.email = Optional.of(person.getEmail().get());
+            this.address = Optional.empty();
+        }
```
Comment (wongkokian) [inline src/main/java/seedu/address/model/person/Person.java]: Same here. Suggestion to use switch statements.

**Review by wongkokian — COMMENTED** [review]
Some suggestions, other than that, LGTM! 👍 :)

File: src/main/java/seedu/address/logic/commands/ListCommand.java (hunk 2 above)
Comment (jay9645) [reply src/main/java/seedu/address/logic/commands/ListCommand.java]: Great suggestion, done!

File: src/main/java/seedu/address/logic/commands/ListCommand.java (hunk 3 above)
Comment (jay9645) [reply src/main/java/seedu/address/logic/commands/ListCommand.java]: Great suggestion, made the change.

File: src/main/java/seedu/address/model/person/Person.java (hunk 5 above)
Comment (jay9645) [reply src/main/java/seedu/address/model/person/Person.java]: Done!

File: src/main/java/seedu/address/logic/parser/ParserUtil.java (hunk 4 above)
Comment (jay9645) [reply src/main/java/seedu/address/logic/parser/ParserUtil.java]: Done!

**Pull request merged**